net/swapnet/host.go

```diff
@@ -0,0 +1,134 @@
+package swapnet
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	logging "github.com/ipfs/go-log"
+	libp2pnetwork "github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/athanorlabs/atomic-swap/common"
+	"github.com/athanorlabs/atomic-swap/common/types"
+	"github.com/athanorlabs/atomic-swap/net"
+)
+
+const (
+	protocolID     = "/atomic-swap/0.1"
+	maxMessageSize = 1 << 17
+)
+
+var log = logging.Logger("host")
+var _ Host = &host{}
+
+// Host represents a peer-to-peer node (ie. a host)
+type Host interface {
```

Comment:
I can't keep these types straight in my head.  Both the `swapnet` package and the `net` package above it have a file named `host.go`. In that file, they both have an interface named `Host` and the comment above both interfaces is 100% identical. Can we reduce the cognitive load with comments and maybe some file/type-name differentiation?

Reply:
okay I can updated the comment, what do you suggest renaming the hosts? should this be `SwapHost` or something (although I feel that's kind of redundant)

Reply:
Would it be correct to say that `net.Host` provides the raw libp2p "node" functionality and that `swapnet.Host` extends the functionality of this "node" to make it a "peer"? Then you could have `net.Node` and `swapnet.Peer` as possible names.

I'm just throwing out ideas. Since I didn't come up with the abstraction, my ideas may be less than perfect. The goal is to have more distinct (and hopefully, intuitive) naming, so readers/developers can quickly keep the pieces separate in their heads and understand how the pieces interact with each other.